src/vmm/src/vstate/vm.rs

```diff
@@ -150,6 +150,10 @@ pub struct Vm {
     // On aarch64 we need to keep around the fd obtained by creating the VGIC device.
     #[cfg(target_arch = "aarch64")]
     irqchip_handle: Option<Box<dyn GICDevice>>,
+
+    // the next available slot to be used for adding a user memory region wit
+    // KVM_SET_USER_MEMORY_REGION
+    next_memory_slot: u16,
```

Comment:
Hm, I'm not sure this is the best place to put this field. Conceptually it has nothing to do with the `Vm` object, which should be just the "gateway" inside the vmfd. An internal counter that's only relevant if one attaches virtio-mem devices doesn't need to be here. I'd put it in the `Vmm` and propagate it as a param.